File: app/src/main/java/com/example/hitchapp/models/Request.java
package com.example.hitchapp.models;

import org.json.JSONArray;
import org.parceler.Parcel;

import com.parse.ParseClassName;
import com.parse.ParseObject;
import com.parse.ParseUser;

@Parcel(analyze={Request.class})
@ParseClassName("Request")
public class Request extends ParseObject{

    public static final String KEY_RIDE = "ride";
    public static final String KEY_REQUESTER = "requester";
    public static final String KEY_DRIVER = "driver";


    public ParseUser getRequester(){ return getParseUser(KEY_REQUESTER); }

    public void setRequester(ParseUser user){
        put(KEY_REQUESTER, user);
    }

    public Ride getRide(){ return (Ride) getParseObject(KEY_RIDE);}

    public void setRide(Ride ride){ put(KEY_RIDE, ride);}

    public ParseUser getDriver(){ return getParseUser(KEY_DRIVER); }

    public void setDriver(ParseUser driver) { put(KEY_DRIVER, driver);} {
    }
}

File: app/src/main/java/com/example/hitchapp/models/Request.kt
package com.example.hitchapp.models

import com.parse.ParseClassName
import com.parse.ParseObject
import com.parse.ParseUser
import org.parceler.Parcel

@Parcel(analyze = [Request::class])
@ParseClassName("Request")
class Request : ParseObject() {
    var requester: ParseUser?
        get() = getParseUser(KEY_REQUESTER)
        set(user) {
            user?.let { put(KEY_REQUESTER, it) }
        }

    var ride: Ride?
        get() = getParseObject(KEY_RIDE) as Ride?
        set(ride) {
            ride?.let { put(KEY_RIDE, it) }
        }

    var driver: ParseUser?
        get() = getParseUser(KEY_DRIVER)
        set(driver) {
            driver?.let { put(KEY_DRIVER, it) }
        }

    companion object {
        const val KEY_RIDE = "ride"
        const val KEY_REQUESTER = "requester"
        const val KEY_DRIVER = "driver"
    }
}